assembly_visualization.md

```diff
@@ -0,0 +1,190 @@
+
+ **IGV**: Integrative Genomics Viewer.
+
+ We will use this software to evaluate our assemblies, among others to view how the reads map to the assembly.
+
+ We will also add tracks with genome annotation and gaps.
+
+We have assembled reads and annotated using Bifrost pipeline. The data for today's practical are found in `/projects/nn9305k/Bioinfo_mapping_training`
+
+## 1. Mapping the reads against a reference/assembly
+
+### 1.1 What is read-mapping
+
+It is simply finding a/the matching locus/area of a read on a sequence. Usually few mismatches are allowed (think about the consequences).
+
+Reads can be mapped as paired or single. If paired is used, then the matching regions are defined by the insert size and the length of each read
+
+<center> <a href="https://commons.wikimedia.org/wiki/File:Mapping_Reads.png"> <img src="https://upload.wikimedia.org/wikipedia/commons/2/2e/Mapping_Reads.png" width="400">
+<br>
+
+
+### 1.2 why mapping reads
+
+1) Statistics are necessary but might not be sufficient to evaluate the quality of an assembly (or to compare different methods used to assemble your reads). Read mapping can help identifying problematic areas.
+
+We want to look at:
+- the coverage regularity (ex: some repeated regions might have increased coverage)
+- the coverage at the beginning and end of scaffolds - is it good enough?
+- are they positions where pileup of reads show polymorphism?
+- ...
+
+2) to identify SNPs: some methods use reads mapping against a reference to identify and type SNPs
+
+3) Assembly polishing software such as `pilon` and `reapr` use mapped reads to identify potential improvement in assemblies. It is good to visualize what information they are using ...
+  > eg. variation in coverage, wrong read pairs orientation, discrepancy between expected insert size and actual insert size obtained from read mapping (ie. longer than expected)
+
+ - [ ] other?
+
+### 1.3 Re-map the reads to the assembly
+
+We use [bwa mem](http://bio-bwa.sourceforge.net/bwa.shtml) option from [bwa tools](http://bio-bwa.sourceforge.net/) software.
+
+You will use the trimmed-filtered reads in: `/projects/nn9305k/Bioinfo_mapping_training/bifrost_run/bbduk_trimmed` and the assembly: `/projects/nn9305k/Bioinfo_mapping_training/bifrost_run/prokka/MiSeq_Ecoli_MG1655_50x.fna`
+
+We could use the assembly output from `Pilon` **BUT** then we would have to create aliases between labels used by `Prokka` and by `Pilon` to be able to visualize the assembly with IGV. `Prokka` assembly is identical to `Pilon` assembly, only labels have been changed.
+
+In your home make a directory for today's work
+and a folder called `mapping` where you will copy the input files
+
+```bash
+# softlink the required input files in the mapping folder:
+ln -s <path>/bifrost_run/bbduk_trimmed/*.fq.gz .
+ln -s <path>/bifrost_run/pilon/MiSeq_Ecoli_MG1655_50x_pilon_spades.fasta .
+
+# we will use Bifrost with the command line
```

Comment:
We will use the software included in the Bifrost pipeline. This is available as a conda environment, called bifrost.

Reply:
added